pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1771,11 +1780,20 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 	}
 
 	if needsSCSIController(vmi) {
-		scsiController := api.Controller{
-			Type:   "scsi",
-			Index:  "0",
-			Model:  InterpretTransitionalModelType(&c.UseVirtioTransitional),
-			Driver: controllerDriver,
+		if isS390X(c.Architecture) {
```

Comment:
Why do we need to distinguish the architecture you are already passing it to the function?

Reply:
The reason here is that then SCSI controller needs `virtio-scsi` whereas `InterpretTransitionalModelType` returns `virtio` for s390x, since it also covers other devices (and `virtio-transitional` as well as `virtio-non-transitional` do not work).